### PR TITLE
New package: nsxiv-27.1

### PR DESCRIPTION
**Pull request by WitherCubes**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by WitherCubes** [comment]
Does anyone have any idea why merging this pull request takes so long?

**Comment by paper42** [comment]
> Does anyone have any idea why merging this pull request takes so long?

New packages need to be properly reviewed and maintainers need to be at least a bit interested in maintaining the package if the original maintainer goes away. If this was just an upstream change for sxiv, it would be simpler to review, but this is a fork which changes the project name, so it needs to be a new package.

I noticed this package installs examples in /usr/share/nsxiv/exec/, but sxiv installs them in /usr/share/examples. Do you know why this change was made? Do you think we should move them to /usr/share/examples?

**Comment by WitherCubes** [comment]
Thanks for the information, I didn't think about placing the examples in /usr/share/example. It is a great idea to move them.

**Comment by N-R-K** [comment]
> I noticed this package installs examples in /usr/share/nsxiv/exec/, but sxiv installs them in /usr/share/examples.

Hi, I don't believe we changed anything. Sxiv installs them in `$(DESTDIR)$(PREFIX)/share/sxiv/exec/` , we only chnaged the name to nsxiv.
https://github.com/muennich/sxiv/blob/1d28627868f5133a25a0197af21a187799ffa98e/Makefile#L78

**Comment by paper42** [comment]
> > I noticed this package installs examples in /usr/share/nsxiv/exec/, but sxiv installs them in /usr/share/examples.
> 
> Hi, I don't believe we changed anything. Sxiv installs them in `$(DESTDIR)$(PREFIX)/share/sxiv/exec/` , we only chnaged the name to nsxiv.
> [muennich/sxiv@`1d28627`/Makefile#L78](https://github.com/muennich/sxiv/blob/1d28627868f5133a25a0197af21a187799ffa98e/Makefile#L78)

Hi, Void patches this in the sxiv package: https://github.com/void-linux/void-packages/blob/75403cef76a93199456befdcb8884fbac39adbb3/srcpkgs/sxiv/patches/sxiv-1.3-makefile.patch. Do you think this could be upstreamed? Having examples in /usr/share/examples is quite common and people expect them there.

**Comment by N-R-K** [comment]
> Having examples in /usr/share/examples is quite common and people expect them there.

Good point, I'll open an issue and see what other maintainers have to say about this. We try to be backwards compatible with `sxiv` as much as possible, so I can't guarantee weather this will get upstreamed.

**Comment by paper42** [comment]
@WitherCubes could you add the patch?

**Comment by N-R-K** [comment]
Btw, since the topic of package maintenance cam up, we ended up carelessly including some non-POSIX extensions in our Makefile, we're in the process of reverting them https://github.com/nsxiv/nsxiv/pull/71.
This should get merged in `v28` and after this, I don't believe our build system will change much if at all. So bumping version should be a smooth experience for maintainers.

**Comment by eylles** [comment]
Altho not all distros follow the usage of `/udr/share/examples` i think probably the sanest way would be to make 2 additions into upstream, have a var for the directory where to install the examples, which would default to the current one (`/usr/share/nsxiv/exec`) but have that var set to `/usr/share/examples` in the void linux build, additionally have another option like EXAMPLES which would be 1 by default and changing ti to 0 would disable installing them, that to also respect the way sxiv is installed in the debian package.

**Comment by N-R-K** [comment]
@eylles That seems like needless complexity. Currently in https://github.com/nsxiv/nsxiv/pull/86 we're offering a `DOCPREFIX`. Distros can change that to their liking.

**Comment by WitherCubes** [comment]
Added the patch, tested it out and it is working.

**Comment by N-R-K** [comment]
> Do you think this could be upstreamed?

Yes, merged https://github.com/nsxiv/nsxiv/pull/86

That will be for `v28` however. For this version, a patch will still be needed.

File: srcpkgs/nsxiv/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	[ -e ${FILESDIR}/config.h ] && cp ${FILESDIR}/config.h config.h ||:
+}
+
+post_install() {
+	vinstall nsxiv.desktop 644 usr/share/applications
+}
+
+if [ "$CROSS_BUILD" ]; then
+	# otherwise ft2build.h is not found
+	hostmakedepends+=" freetype-devel"
```
[the remark this answers is not in the document]
Comment (WitherCubes) [reply srcpkgs/nsxiv/template]: Fixed it, see [this commit](https://github.com/void-linux/void-packages/pull/33009/commits/c21465ff936c8a7ae96d3631ab3771aa486b22b4).

**Comment by paper42** [comment]
Squash the commits and reset the revision to 1.

**Comment by WitherCubes** [comment]
Some checks are failing, @paper42 do you know the details of this problem?

File: srcpkgs/nsxiv/template [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+pre_build() {
+	[ -e ${FILESDIR}/config.h ] && cp ${FILESDIR}/config.h config.h ||:
+}
+
+post_install() {
+	vinstall nsxiv.desktop 644 usr/share/applications
+}
+
+CFLAGS="${XBPS_CROSS_BASE}/usr/include/freetype2"
```
Comment (paper42) [inline srcpkgs/nsxiv/template:22]:
```suggestion
CFLAGS="-I${XBPS_CROSS_BASE}/usr/include/freetype2"
```
and move this up to the end of the first block (after checksum)

**Comment by WitherCubes** [comment]
@paper42 Should I bother about the lint template check failing or ignore it and squash the commits?

**Pull request merged**